### PR TITLE
CORDA-2998 fix network builder

### PR DESCRIPTION
**Pull request by szymonsztuka**
Network-bootstrapper for Docker/Azure renamed to Network-Builder (by Anthony).
Fix netowrk-builder to eork with Corda 4 (by Stefano).
Reusing Corda official Docker image as a base image + minor fixes (by Szymon).

Manually testsed by ``./gradlew tools:network-builder:publishToMavenLocal``
and then using ``~/.m2/repository/net/corda/corda-tools-network-builder/5.0-SNAPSHOT/corda-tools-network-builder-5.0-SNAPSHOT.jar``.
Following  the updated docs: https://github.com/corda/corda/blob/5eaa7262a340415369f043b9442143e15c769571/docs/source/network-builder.rst
and tested both GUI with locla Docker and Azure service. tests command line as well.

### CODE REVIEW

File: docs/source/network-builder.rst [hunk 1]
```diff
@@ -8,11 +8,15 @@ containers to abstract the complexity of managing a distributed network away fro
 
 .. image:: _static/images/network-builder-v4.png
 
-The network you build will either be made up of local ``docker`` nodes *or* of nodes spread across Azure
-containers. More backends may be added in future. The tool is open source, so contributions to add more
+The network you build will either be made up of local ``Docker`` nodes *or* of nodes spread across Azure
+containers.
+For each node a separate Docker image is built based on `corda/corda-zulu-4.0 <https://hub.docker.com/r/corda/corda-zulu-4.0>`_.
+Unlike the official image, a `node.conf` file and CorDapps are embedded into an image
```
Comment (anthonykeenan) [inline docs/source/network-builder.rst]: into the image

**Comment by anthonykeenan** [comment]
This will fix https://github.com/corda/corda/issues/5203 (after backport to v4)

File: docs/source/network-builder.rst [hunk 2]
```diff
@@ -95,45 +102,50 @@ You can interact with the nodes by SSHing into them on the port that is mapped t
 
     >>> run networkMapSnapshot
     [
-      { "addresses" : [ "partya0:10020" ], "legalIdentitiesAndCerts" : [ "O=PartyA, L=London, C=GB" ], "platformVersion" : 3, "serial" : 1532701330613 },
-      { "addresses" : [ "notary0:10020" ], "legalIdentitiesAndCerts" : [ "O=Notary, L=London, C=GB" ], "platformVersion" : 3, "serial" : 1532701305115 },
-      { "addresses" : [ "partyc0:10020" ], "legalIdentitiesAndCerts" : [ "O=PartyC, L=Paris, C=FR" ], "platformVersion" : 3, "serial" : 1532701331608 },
-      { "addresses" : [ "partyb0:10020" ], "legalIdentitiesAndCerts" : [ "O=PartyB, L=New York, C=US" ], "platformVersion" : 3, "serial" : 1532701330118 }
+      { "addresses" : [ "partya0:10020" ], "legalIdentitiesAndCerts" : [ "O=PartyA, L=London, C=GB" ], "platformVersion" : 4, "serial" : 1532701330613 },
```
Comment (anthonykeenan) [inline docs/source/network-builder.rst]: Is there a |platform_verison| docs constant or something we can use?

File: docs/source/network-builder.rst (hunk 1 above)
Comment (szymonsztuka) [reply docs/source/network-builder.rst]: fixed

File: docs/source/network-builder.rst (hunk 2 above)
Comment (szymonsztuka) [reply docs/source/network-builder.rst]: fied

File: docs/source/network-builder.rst [hunk 3]
```diff
@@ -95,45 +102,50 @@ You can interact with the nodes by SSHing into them on the port that is mapped t
...
 
 The Corda Network Builder also provides a GUI for when automated interactions are not required. To launch it, run
-``java -jar <path/to/network-builder-jar> -g``.
+``java -jar <path/to/corda-tools-network-builder.jar> -g``.
 
 Starting the nodes
 ^^^^^^^^^^^^^^^^^^
 
-1. Click ``Open nodes ...`` and select the folder where you built your nodes in :ref:`creating_the_base_nodes` and
+1. Click ``Open nodes ...`` and select jar the folder where you built your nodes in :ref:`creating_the_base_nodes` and
```
Comment (roastario) [inline docs/source/network-builder.rst]: Select jar? Does this mean the Corda jar?

File: tools/network-builder/src/main/resources/node-Dockerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
+FROM corda/corda-zulu-4.0
+
+# Copy corda files
+ADD --chown=corda:corda corda.jar               /opt/corda/corda.jar
```
Comment (roastario) [inline tools/network-builder/src/main/resources/node-Dockerfile]: Should we allow them to overwrite the corda jar? ( I don't know )

File: tools/network-builder/src/main/kotlin/net/corda/networkbuilder/nodes/NodeFinder.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package net.corda.networkbuilder.nodes
+
+import com.typesafe.config.ConfigException
+import com.typesafe.config.ConfigFactory
+import net.corda.networkbuilder.Constants
+import net.corda.core.utilities.contextLogger
+import java.io.File
+
+class NodeFinder(private val scratchDir: File) {
```
Comment (roastario) [inline tools/network-builder/src/main/kotlin/net/corda/networkbuilder/nodes/NodeFinder.kt]: I know it's my fault, but is there any way to combine with the NotaryFinder?

File: docs/source/network-builder.rst (hunk 3 above)
Comment (szymonsztuka) [reply docs/source/network-builder.rst]: fixed

File: tools/network-builder/src/main/resources/node-Dockerfile (hunk 4 above)
Comment (szymonsztuka) [reply tools/network-builder/src/main/resources/node-Dockerfile]: Indeed, remove this and modify bash script to acces corda.jar from `bin` as the base image places the jar

File: tools/network-builder/src/main/kotlin/net/corda/networkbuilder/nodes/NodeFinder.kt (hunk 5 above)
Comment (szymonsztuka) [reply tools/network-builder/src/main/kotlin/net/corda/networkbuilder/nodes/NodeFinder.kt]: Combained NotaryFider into NodeFinder, with two public methods `findNodes`, `findNotaries`

**Pull request merged**